lib/cli/index.js

```diff
@@ -7,6 +7,10 @@ var logger        = require('heimdalljs-logger')('ember-cli:cli/index');
 var merge         = require('ember-cli-lodash-subset').merge;
 var path          = require('path');
 
+// work around misbehaving libraries, so we can correctly cleanup before
+// actually exiting.
+require('capture-exit').captureExit();
```

Comment:
Note why I'm not concerned about this: even if things were to be interrupted before this point it doesn't suffer the negative consequences that we are trying to prevent.